File: api-gateway/src/chat/dto/create-conversation.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsString, MinLength } from 'class-validator';

export class CreateConversationDto {
  @ApiProperty({
    example: 'p-123',
    description: 'ID de dominio del paciente.',
  })
  @IsString()
  @MinLength(1)
  patientId: string;

  @ApiProperty({
    example: 'd1',
    description: 'ID de dominio del dentista.',
  })
  @IsString()
  @MinLength(1)
  dentistId: string;
}

File: chat-service-csharp/ChatService/DTOs/CreateConversationDto.cs
using System.ComponentModel.DataAnnotations;

namespace ChatService.DTOs;

public class CreateConversationDto
{
    [Required]
    [MinLength(1)]
    public string PatientId { get; set; } = string.Empty;

    [Required]
    [MinLength(1)]
    public string DentistId { get; set; } = string.Empty;
}
